controllers/cloud.redhat.com/providers/providers.go

```diff
@@ -157,16 +157,28 @@ func createResource(cache *rc.ObjectCache, resourceIdent rc.ResourceIdent, nn ty
 	err = cache.Create(resourceIdent, nn, nobj)
 
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error creating object: %s%s %w", nn.Name, nn.Namespace, err)
 	}
 	return nobj, nil
 }
 
 func updateResource(cache *rc.ObjectCache, resourceIdent rc.ResourceIdent, object client.Object) error {
 	err := cache.Update(resourceIdent, object)
 
+	gvks, nok, werr := cache.GetScheme().ObjectKinds(resourceIdent.GetType())
+
+	if werr != nil {
```

Comment:
Could we have a more descriptive name or a comment? I don't know what this failure condition is.